modules/php/Cards/Rules/RuleMysteryPlay.php

```diff
@@ -9,22 +9,44 @@ public function __construct($cardId, $uniqueId)
   {
     parent::__construct($cardId, $uniqueId);
 
-    $canBeUsedByPlayer = true;
-
     $this->name = clienttranslate("Mystery Play");
     $this->subtitle = clienttranslate("Free Action");
     $this->description = clienttranslate(
       "Once during your turn, you may take the top card from the draw pile and play it immediately."
     );
   }
 
+  public function canBeUsedInPlayerTurn($player_id)
+  {
+    return Utils::playerHasNotYetUsedMysteryPlay();
+  }
+
   public function immediateEffectOnPlay($player)
   {
-    // @TODO
+    // nothing
   }
 
   public function immediateEffectOnDiscard($player)
   {
     // nothing
   }
+
+  public function freePlayInPlayerTurn($player_id)
+  {
+    $game = Utils::getGame();
+    $game->setGameStateValue("playerTurnUsedMysteryPlay", 1);
+
+    // draw top card (this is moved to hand automatically)
+    $card = $game->cards->pickCard("deck", $player_id);
+    $game->notifyPlayer($player_id, "cardsDrawn", "", [
+      "cards" => [$card],
+    ]);
+
+    // And we mark it as the next "forcedCard" to play
+    $game->setGameStateValue("forcedCard", $card["id"]);
+
+    // But Mystery Play should not really be counted for play rule
+    $alreadyPlayed = $game->getGameStateValue("playedCards");
```

Comment:
Oh, good catch. I think there is a bug anywhere else in the code where we use "forcedCard". We should actually do this at `PlayCardTrait.php:22`

Reply:
Aha, I actually thought this was a special situation for the Mystery Play.
But indeed, we should check the other forced cards, and most likely none of them should be added as a "play" count.
Then we can apply this generic in PlayCardTrait.

Reply:
Ok, checked and applied generically for all forced plays.